core/invoker/src/main/scala/whisk/core/containerpool/ContainerProxy.scala

```diff
@@ -336,7 +362,8 @@ class ContainerProxy(
         // and do not block further execution of the future. They are completely
         // asynchronous.
         activation.andThen {
-            case Success(activation) => sendActiveAck(tid, activation)
+            // the activation future will always complete with Success
+            case Success(ack) => sendActiveAck(tid, ack)
```

Comment:
I was tempted here to assert the invariant as:

`case result => val Success(ack)` forcing an error should the contract be vioated